msteams-platform/messaging-extensions/api-based-overview.md

```diff
@@ -8,79 +8,574 @@ ms.author: anclear
+
+<br>
+
+:::image type="content" source="../assets/images/Copilot/api-based-me-flow-2.png" alt-text="Screenshot shows the query flow between a user, Teams Client, and Teams bot service using API Message Extensions. The diagram also shows how the API spec, the rendering templates, the API relate to each other." lightbox="../assets/images/Copilot/api-based-me-flow-2.png":::
+*User query flow using API Message Extensions. There's no need for a developer maintained handler service as long as the interaction is clearly outlined in the OpenAPI Specification in the App Package.*
+
+
+<br>
+<br>
+
+Here's a high-level sequence of events that occur during a query command invocation:
+
+1. When a user invokes a query command, the parameters of the query command are received by the Teams Bot Service.
+
+1. The query command is defined inside the app manifest file. The command definition contains a reference to the operation_id inside the OpenAPI Specification file along with the details of the parameters that the Teams client renders for that command. For reference, the operation_id inside the OpenAPI Specification file is unique to a particular HTTP operation.
+
+1. The Teams Bot Service then uses the parameters supplied by the user along with the copy of the OpenAPI Specification for the associated operation_id to build an HTTP request for the developer’s endpoint.
+
+1. If authentication is required and is configured in the manifest. It's resolved to the appropriate token or key. This token or key is used as part of the outgoing request. *[Optionally]*
+
+1. The Teams bot service performs the HTTP request to the developer’s service.
+
+1. The developer’s service should respond in accordance with the schema outlined in the OpenAPI Specification. This is in JSON format.
+
+1. The Teams client must show the results back to the user. To convert the JSON results from the previous step to UI, the Teams bot service uses the response Rendering template to build an Adaptive Card for each result.
+
+1. The Adaptive Cards are sent to the client, which renders them in the UI.
+
+:::image type="content" source="../assets/images/Copilot/api-based-me-query-sequence-diagram.png" alt-text="Diagram shows the high-level sequence flow when a query is invoked in an API-based message extension." lightbox="../assets/images/Copilot/api-based-me-query-sequence-diagram.png":::
+
+
 ## Prerequisites
 
-Before you get started, ensure that you adhere to the following requirements:
+The app definition package includes various compelling artifacts that support the functionality of this feature. Before you get started, ensure that have a basic understanding of the following files:
 
 > [!div class="checklist"]
 >
-> * [OpenAPI Description (OAD)](#openapi-description)
-> * [Update app manifest](#update-app-manifest)
+> * [OpenAPI Description (OAD)](#openapi-description-oad)
```

Comment:
Can you include the simple purpose of each of these as it is in the doc? 
![image](https://github.com/user-attachments/assets/64469d72-fd7c-4f2a-98dd-07b2305a1a6d)

Reply:
Updated in the Overview page for each section under [prerequisites ](https://review.learn.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/api-based-overview?branch=pr-en-us-10743#prerequisites).